### PR TITLE
Add intro to Docker in tutorials

### PR DESCRIPTION
**Pull request by xavierlepretre**
Documentation content update for tutorials.

This PR adds a page to give a quick intro on Docker to beginners. It also adds links to it in the content.

### R1: Internal review (B9lab)

- Technical review
  - [x] requested
  - [x] completed
- Language review
  - [x] requested
  - [x] completed

### R2: External review

- Technical review
  - [x] requested
  - [x] completed
- Language review
  - [ ] requested
  - [ ] completed

### R3: Internal QA review (B9lab)

- Technical review
  - [ ] requested
  - [ ] completed
- Language review
  - [ ] requested
  - [ ] completed

### R4: Final external QA review

- Technical review
  - [ ] requested
  - [ ] completed
- Language review
  - [ ] requested
  - [ ] completed

### RC: Release Candidate

- [ ] Ready to be merged

### CODE REVIEW

File: tutorials/5-docker-intro/index.md
Comment (AndrewB9lab) [inline tutorials/5-docker-intro/index.md:16]: @xavierlepretre Can we qualify "virtualization"?

"Docker is a kind of XXXXXXXX virtualization with performance benefits."

File: tutorials/5-docker-intro/index.md
Comment (AndrewB9lab) [inline tutorials/5-docker-intro/index.md:20]: Different from what?

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [reply tutorials/5-docker-intro/index.md:16]: Yes

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [reply tutorials/5-docker-intro/index.md:20]: In the field of computer virtualisation, being different from classical virtualisation...

something like that...

File: tutorials/5-docker-intro/index.md
Comment (AndrewB9lab) [inline tutorials/5-docker-intro/index.md:153]: @xavierlepretre - apologies for stepping outside of language review territory! But on this line, is it possible that **-i** should be **-it**?

I ask because the instructions/changes for lines 162-167 don't specify precisely what is being changed, and I see two differences; whereas 176-179 does specify one precise change, removing the "-t".

File: tutorials/5-docker-intro/index.md
Comment (AndrewB9lab) [inline tutorials/5-docker-intro/index.md:237]: The practice is to add `--rm` ... because this automatically **removes** the container? Y/"N? And if so, when is the container removed?

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [reply tutorials/5-docker-intro/index.md:153]: Yep, please put the `t`. From my testing, there is no difference. Not sure why there was none here.

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [reply tutorials/5-docker-intro/index.md:237]: Yes it removes.
The container is removed at the moment it is stopped and exited.

**Review by AndrewB9lab — APPROVED** [review]
Primary language review completed.

File: tutorials/5-docker-intro/index.md
Comment (julienrbrt) [inline tutorials/5-docker-intro/index.md:41]:
```suggestion
This introduces an interesting concept whereby [a machine is described as a file](https://nitter.net/FrancescoCiull4/status/1509458241524224005). This is very useful for reproducibility and DevOps. Further, the images are optimized so that each image is a _diff_ of a parent one.
```

:trollface:

**Review by julienrbrt — APPROVED** [review]
utACK. I have not try the commands but it lgtm.

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [reply tutorials/5-docker-intro/index.md:41]: What's the advantage of nitter?

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [inline tutorials/5-docker-intro/index.md:45]: I think you changed the meaning. I want to say:

> There exist registries that...

You changed it to

> We all know there are registries and some store and/or reference them.

The problem is that it is not obvious that there would be registries. And storing or referencing them is all they do anyway, some only reference, some do both. So it is not like some registries do nothing, others store/reference and others do more than that.

If changing, I would go with a variation on:

> There exist registries that ...

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [inline tutorials/5-docker-intro/index.md:76]:
```suggestion
`-it` is short for `--interactive --tty` and means "with input and output", instead of a fully detached container. Learn more with `docker run --help`. This should return you a Node.js prompt:
```

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [inline tutorials/5-docker-intro/index.md:251]:
```suggestion
<HighlightBox type="best-practice">
```

**Comment by coldice** [comment]
@xavierlepretre just wanted to merge this - please let me know when ready for merge.

File: tutorials/5-docker-intro/index.md
Comment (CitMC) [reply tutorials/5-docker-intro/index.md:45]: @xavierlepretre feel free to change back. 😉

File: tutorials/5-docker-intro/index.md
Comment (xavierlepretre) [inline tutorials/5-docker-intro/index.md:45]:
```suggestion
There exist registries that store and/or reference them, like your regular package manager. [Docker Hub](https://hub.docker.com) is the main one, and your local Docker knows how to download directly from it.
```

**Pull request merged**
